### PR TITLE
Implement bitwise XOR operator (`#`)

### LINKED ISSUE #3420: Support bitwise XOR operator (`#`)
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
See usecase from @FauxFaux on https://github.com/apache/arrow-datafusion/issues/1619

The only remaining bitwise operator left to go is bitwise XOR (`#`) which is tracked by this ticket

**Describe the solution you'd like**
Implement  bitwise XOR operator (following the model of bitwise AND `&` and OR `|`) -- see PRs linked from https://github.com/apache/arrow-datafusion/issues/1619

I think this is a good first issue as there are existing operator examples which can be followed and newcomers could learn the codebase

### PR DESCRIPTION
**Pull request by askoa**
# Which issue does this PR close?

Closes #3420 

# Are there any user-facing changes?
I think inclusion of bitwise XOR is a user-facing change.

Not sure if any documentation change is required. I mimicked a similar PR #1876. If any documentation update is required, then reviewers can point me to the documentation changes required.

### CODE REVIEW

File: datafusion/physical-expr/src/expressions/binary.rs [hunk 1]
```diff
@@ -1295,6 +1298,18 @@ mod tests {
...
+            Int16Array,
+            DataType::Int16,
+            vec![3i16, 2i16, 3i16],
+            Int64Array,
+            DataType::Int64,
+            vec![10i64, 6i64, 5i64],
+            Operator::BitwiseXor,
+            Int64Array,
+            DataType::Int64,
+            vec![9i64, 4i64, 6i64]
```
Comment (alamb) [inline datafusion/physical-expr/src/expressions/binary.rs:1311]: 👍  I verified using  https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=9c79637433e2b569933e58d087d2acb4

**Review by alamb — APPROVED** [review]
Thank you @askoa  -- this is a great first contribution

In terms of documentation it seems we are missing all the bitwise operators  in https://github.com/apache/arrow-datafusion/blob/master/docs/source/user-guide/expressions.md

perhaps you could it (it would be fine to do as a follow on PR)

**Comment by askoa** [comment]
> perhaps you could it (it would be fine to do as a follow on PR)

I created a issue to track this. I will update the documents.

**Review by andygrove — APPROVED** [review]

**Comment by alamb** [comment]
Thanks again @askoa

**Pull request merged**